packages/components/src/focal-point-picker/grid.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GridView,
+	GridLineX,
+	GridLineY,
+} from './styles/focal-point-picker-style';
+
+export default function FocalPointPickerGrid( {
+	bounds = {},
+	percentages = {
+		x: 0.5,
+		y: 0.5,
+	},
+	...props
+} ) {
+	const percentageValue = percentages.x + percentages.y;
+	const animationProps = useRevealAnimation( percentageValue );
+	const style = {
+		width: bounds.width,
+		height: bounds.height,
+	};
+
+	return (
+		<GridView
+			{ ...props }
+			{ ...animationProps }
+			className="components-focal-point-picker__grid"
+			style={ style }
+		>
+			<GridLineX style={ { top: '33%' } } />
+			<GridLineX style={ { top: '66%' } } />
+			<GridLineY style={ { left: '33%' } } />
+			<GridLineY style={ { left: '66%' } } />
+		</GridView>
+	);
+}
+
+/**
+ * Custom hook that renders the "flash" animation whenever the value changes.
+ *
+ * @param {string} value Value of (box) side.
+ */
+function useRevealAnimation( value ) {
+	const [ isActive, setIsActive ] = useState( false );
+	const valueRef = useRef( value );
+	const timeoutRef = useRef();
+
+	const clearTimer = () => {
+		if ( timeoutRef.current ) {
+			window.clearTimeout( timeoutRef.current );
+		}
+	};
+
+	useEffect( () => {
+		if ( value !== valueRef.current ) {
+			setIsActive( true );
+			valueRef.current = value;
+
+			clearTimer();
+
+			timeoutRef.current = setTimeout( () => {
+				setIsActive( false );
+			}, 600 );
+		}
+
+		return () => clearTimer();
+	}, [ value ] );
```

Comment:
Seems like this could be simplied to just:

```
useEffect( () => {
 	setIsActive( true );
 	const timeout = setTimeout( () => {
 		setIsActive( false );
 	}, 600 );

 	return () => window.clearTimeout( timeout );
 }, [ value ] );
```

and avoid all refs. Am I missing anything?

Reply:
That works. I initially created refs so that the animation would not flash on first render. I was able to combine your solution with the `useUpdateEffect` hook from another PR:

https://github.com/WordPress/gutenberg/blob/8a35791d60f8a69e2d155990e9c7e564248a4713/packages/components/src/utils/use-update-effect.js